python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -73,6 +78,8 @@ def __init__(self,
         self._num_workers = num_workers
         self._num_cpus_per_worker = num_cpus_per_worker
         self._num_gpus_per_worker = num_gpus_per_worker
+        self._max_failures = env_integer(MAX_FAILURE_RETRIES, 3)
```

Comment:
can we expose this parameter to the user?

Reply:
This is exposed via an environment variable, which I think should be fine. I wanted to avoid polluting `Trainer` args too much.

Reply:
Having this on by default and hard to configure doesn’t actually make sense, you often want to explicitly debug the workload and this would cause undesirable restarts

Reply:
Ok I'll make the change, but are you suggesting to disable by default?

Reply:
Made the change